server/remoting/bolt/src/main/java/com/alipay/sofa/registry/remoting/bolt/BoltClient.java

```diff
@@ -32,252 +43,220 @@
-                    ret = true;
-                    break;
-                }
-            }
-        } else {
-            //has no channels
-            return true;
-        }
-        return ret;
+        return closed.get();
     }
 
     @Override
-    public void sendOneway(Channel channel, Object message) {
-        if (channel != null && channel.isConnected()) {
-            if (channel instanceof BoltChannel) {
-                BoltChannel boltChannel = (BoltChannel) channel;
-                try {
-                    boltClient.oneway(boltChannel.getConnection(), message);
-                } catch (RemotingException e) {
-                    LOGGER.error("Bolt Client oneway request RemotingException! target url: {}",
-                        boltChannel.getRemoteAddress(), e);
-                }
-            }
+    public Object sendSync(URL url, Object message, int timeoutMillis) {
+        try {
+            RpcClient rpcClient = getNextRpcClient();
+            return rpcClient.invokeSync(createBoltUrl(url), message, timeoutMillis);
+        } catch (RemotingException e) {
+            String msg = "Bolt Client sendSync message RemotingException! target url:" + url;
+            LOGGER.error(msg, e);
+            throw new RuntimeException(msg, e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(
+                "Bolt Client sendSync message InterruptedException! target url:" + url, e);
         }
     }
 
     @Override
     public Object sendSync(Channel channel, Object message, int timeoutMillis) {
-
         if (channel != null && channel.isConnected()) {
-            if (channel instanceof BoltChannel) {
-                BoltChannel boltChannel = (BoltChannel) channel;
-                try {
-                    return boltClient.invokeSync(boltChannel.getConnection(), message,
-                        timeoutMillis);
-                } catch (RemotingException e) {
-                    LOGGER.error("Bolt Client sendSync message RemotingException! target url:"
-                                 + boltChannel.getRemoteAddress(), e);
-                    throw new RuntimeException("Bolt Client sendSync message RemotingException!", e);
-                } catch (InterruptedException e) {
-                    LOGGER.error("Bolt Client sendSync message InterruptedException! target url:"
-                                 + boltChannel.getRemoteAddress(), e);
-                    throw new RuntimeException(
-                        "Bolt Client sendSync message InterruptedException!", e);
-                }
-            } else {
-                throw new IllegalArgumentException("Input channel instance error! instance class:"
-                                                   + channel.getClass().getName());
+            BoltChannel boltChannel = (BoltChannel) channel;
+            try {
+                RpcClient rpcClient = getNextRpcClient();
```

Comment:
Let's fix it here. In this case, if you specify a channel, you don't want to randomly select a link to call.